_posts/2021-07-05-newsletter.md

```diff
@@ -0,0 +1,199 @@
+- [Formatter](https://github.com/mhartington/formatter.nvim) -- A plugin for asynchronously executing external formatting tools on the current buffer or range.
+- [Hop.nvim](https://github.com/phaazon/hop.nvim) - Easy-motion without messing with your buffer.
+- [Neogit](https://github.com/TimUntersberger/neogit) - Magit but for Neovim.
+
+Not all of these are actually written in Lua: there are many other languages (some of which are typed) that compile to Lua, e.g.,
+- [Fennel(Lisp)](https://fennel-lang.org) via [Aniseed],(https://github.com/Olical/aniseed) or [Fennel-nvim](https://github.com/jaawerth/fennel-nvim),
+- [Teal](https://github.com/teal-language/tl) (a typed dialect of Lua),
+- [Amulet ML](https://amulet.works),
+- [MoonScript](https://github.com/leafo/moonscript) via [nvim-moonmaker](https://github.com/svermeulen/nvim-moonmaker) ,
+- [Vim9script](https://github.com/tjdevries/vim9jit) (you read that right).
+
+### Lua config
+
+It is also possible to write user configuration in Lua: If there is an `init.lua`, it is read _instead of_ `init.vim` (these cannot coexist, and having both in your config directory will give an error), and `.lua` files in runtime directories (`plugin/`, `colorscheme/`, `after/` etc.) are sourced in addition to (after) Vimscript files. (Yes, this means that you can write your color scheme as a `.lua` file!) Note that this is entirely optional and not needed to enjoy the new functions introduced in Neovim 0.5; furthermore, not every Vimscript configuration option has a fully native Lua equivalent yet. Extending the native API to cover these as well is part of the goal for Neovim 0.6.
+
+For a comprehensive and up-to-date guide on using Lua for Neovim scripting and configuration, see  [Getting started using Lua in Neovim](https://github.com/nanotee/nvim-lua-guide). A good model for a Lua configuration using `init.lua` is [Defaults.nvim](https://github.com/mjlbach/defaults.nvim).
+
+
+## Language Server Protocol (LSP)
+
+The [Language Server Protocol](https://microsoft.github.io/language-server-protocol/) (LSP) is an open, JSON-RPC-based protocol for communication between code editors and language servers, which provide programming language-specific features such as
+
+- completion,
+- hover/tooltips,
+- go to definition,
+- show/go to references,
+- show method signatures,
+- rename,
+- code actions (automatic formatting, organize imports, ...),
+
+and more.
+
+The idea is to separate these features into an editor-independent but language-specific server and a language-independent but editor-specific client, which communicate via the Language Server Protocol over RPC. (It should be pointed out that not every server implements every feature, and the quality of the responses can vary wildly. The "reference implementation" in VS Code also often adds non-standard features that are not covered by the LSP itself).
```

Comment:
`over RPC. ` remove dot

Reply:
No, that is correct here -- there are two separate sentences in the parenthesis.

Reply:
But the final dot needs to be inside the parenthesis.